apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -164,6 +164,27 @@
     ST =:= ?status_connecting; ST =:= ?status_connected; ST =:= ?status_disconnected
 ).
 
+%% If owner_id exists, log owner ID, otherwise log resource_id
+-define(LOG(LEVEL, FIELDS, DATA),
+    case maps:get(owner_id, DATA#data.opts, undefined) of
+        OWNER_ID when is_binary(OWNER_ID) ->
+            ?SLOG(
+                LEVEL,
+                maps:merge(FIELDS, #{
+                    owner_id => binary_to_list(OWNER_ID),
+                    internal_resid => binary_to_list(DATA#data.id)
+                }),
+                #{
+                    tag => tag(DATA#data.group, DATA#data.type)
+                }
+            );
+        _ ->
+            ?SLOG(LEVEL, maps:merge(FIELDS, #{resource_id => binary_to_list(DATA#data.id)}), #{
+                tag => tag(DATA#data.group, DATA#data.type)
+            })
+    end
+).
```

Comment:
Q: couldn't we just do something like `logger:set_process_metadata(#{...})` and this extra info would get logged (at least by the JSON formatter)?